idaes/generic_models/properties/core/coolprop/tests/test_coolprop_wrapper.py

```diff
@@ -0,0 +1,934 @@
+            "state_bounds": {"flow_mol": (0, 100, 1000, pyunits.mol/pyunits.s),
+                             "temperature": (273.15, 300, 500, pyunits.K),
+                             "pressure": (5e4, 1e5, 1e6, pyunits.Pa)},
+            "pressure_ref": (101325, pyunits.Pa),
+            "temperature_ref": (298.15, pyunits.K),
+
+            "parameter_data": {"PR_kappa": {("benzene", "benzene"): 0.000}}}
+
+        m.fs.props = GenericParameterBlock(default=configuration)
+
+        m.fs.state = m.fs.props.build_state_block(
+            [0], default={"defined_state": True})
+
+        m.fs.state[0].flow_mol.fix(1)
+        m.fs.state[0].pressure.fix(101325)
+        m.fs.state[0].temperature.fix(300)
+        m.fs.state[0].mole_frac_comp["benzene"].fix(1)
+
+        return m
+
+    @pytest.mark.unit
+    def test_physical_constants(self, m):
+        # Benzene parameters
+        assert isinstance(m.fs.props.benzene.temperature_crit, Var)
+        assert m.fs.props.benzene.temperature_crit.fixed
+        assert value(m.fs.props.benzene.temperature_crit) == CoolProp.PropsSI(
+            "TCRIT", "Benzene")
+
+        assert isinstance(m.fs.props.benzene.pressure_crit, Var)
+        assert m.fs.props.benzene.pressure_crit.fixed
+        assert value(m.fs.props.benzene.pressure_crit) == CoolProp.PropsSI(
+            "PCRIT", "Benzene")
+
+        assert isinstance(m.fs.props.benzene.dens_mol_crit, Var)
+        assert m.fs.props.benzene.dens_mol_crit.fixed
+        assert value(m.fs.props.benzene.dens_mol_crit) == CoolProp.PropsSI(
+            "RHOMOLAR_CRITICAL", "Benzene")
+
+        assert isinstance(m.fs.props.benzene.mw, Param)
+        assert value(m.fs.props.benzene.mw) == CoolProp.PropsSI(
+            "molarmass", "Benzene")
+
+        assert isinstance(m.fs.props.benzene.omega, Var)
+        assert m.fs.props.benzene.omega.fixed
+        assert value(m.fs.props.benzene.omega) == CoolProp.PropsSI(
+            "acentric", "Benzene")
+
+    @pytest.mark.unit
+    def test_dens_mol_liq_comp(self, m):
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n1, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n2, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n3, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n4, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n5, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_n6, Var)
+
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t1, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t2, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t3, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t4, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t5, Var)
+        assert isinstance(m.fs.props.benzene.dens_mol_liq_comp_coeff_t6, Var)
+
```

Comment:
One of the things I worry about is if we will detect / catch if the CoolProp functional form or units change.  Does this test cover the case (at least for Benzene), as the numbers will change if the form or the units change?

Do we need similar tests for other components, too?

Reply:
That depends on how subtle the change is. At the moment we are lucky in that CoolProp has a relatively limited set of forms that t uses for any given property (and all components use the same for for a given property).

So, it would depend on how CoolProp made the change. If they added a new form with a different name, we would catch it easily. If they made a change to an existing form but kept the current names, we would have more difficulty. With luck, we would see failures due to changes in the parameters (number of coefficients or values), or at worst numerical differences when we validate our calculations against CoolProp. But if we were unlucky, it might slip through our notice...